c/meterpreter/source/extensions/priv/namedpipe.h

```diff
@@ -1,7 +1,14 @@
 #ifndef _METERPRETER_SOURCE_EXTENSION_PRIV_PRIV_SERVER_ELEVATE_TECHNIQUES_NAMEDPIPE_H
 #define _METERPRETER_SOURCE_EXTENSION_PRIV_PRIV_SERVER_ELEVATE_TECHNIQUES_NAMEDPIPE_H
 
-DWORD elevate_via_service_namedpipe( Remote * remote, Packet * packet );
-DWORD elevate_via_service_namedpipe2( Remote * remote, Packet * packet );
+typedef DWORD(*PostImpersonationCallback)(LPVOID);
+DWORD THREADCALL elevate_namedpipe_thread(THREAD* thread);
+DWORD elevate_via_service_namedpipe(Remote* remote, Packet* packet);
+DWORD elevate_via_service_namedpipe2(Remote* remote, Packet* packet);
+
+typedef struct _PRIV_POST_IMPERSONATION {
+	PostImpersonationCallback pCallback;
+	PVOID                     pCallbackParam;
+} PRIV_POST_IMPERSONATION, * PPRIV_POST_IMPERSONATION;
```

Comment:
The `elevate_namedpipe_thread` function now takes a callback to execute after the impersonation takes place. This allows the function to be reused for the new RPCSS variant while maintaining compatibility with the original 2 service-based techniques. The original two use `post_callback_use_self` which just updates the meterpreter token to the new value which is the SYSTEM token, while the RPCSS variant uses a new one.